### PR TITLE
Automationspn

### LINKED ISSUE #25: [TASK 1] Implement script to create Azure Function for /users API
This task is to create a PowerShell script to create Azure Function resources from an ARM template.

### PR DESCRIPTION
**Pull request by selinapn**

### CODE REVIEW

File: automationspn/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,137 @@
+{
```
Comment (toddysm) [inline automationspn/azuredeploy.json]: The folder structure we discussed was:
```
/ (root)
/automation
/automation/functions
```
the `azuredeploy.json` and the `script.ps1` files should be in the `/automation/functions` folder. Structuring your files in an intuitive and logical way will help you better manage the source code when it grows and when you have more people and teams involved.

File: automationspn/script.ps1 [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#Create a local function project
+#Choose folder name
+$foldername = Read-Host -Prompt "Enter a name for your folder" 
+
+#create local function project in powershell
+func init $foldername --powershell
```
Comment (toddysm) [inline automationspn/script.ps1]: This part actually creates a function (and generates code) on your local machine. This is a separate task. You should remove that

File: automationspn/script.ps1 [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+cd $foldername
+
+#create unique name for HTTP Trigger function
+$functionname = Read-Host -Prompt "Name your function"
+#add function to the project
+func new --name $functionname --template "HTTP trigger" --authlevel "anonymous" 
+
+
+$resourceGroupName = Read-Host -Prompt "Name your resource group"
+$location = 'West US 2'
```
Comment (toddysm) [inline automationspn/script.ps1]: Hard coding the location is not recommended. What if the automation is used to create a function in a different location.

File: automationspn/script.ps1 [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+#create local function project in powershell
+func init $foldername --powershell
+
+#navigate into the project folder
+cd $foldername
+
+#create unique name for HTTP Trigger function
+$functionname = Read-Host -Prompt "Name your function"
+#add function to the project
+func new --name $functionname --template "HTTP trigger" --authlevel "anonymous"
```
Comment (toddysm) [inline automationspn/script.ps1]: Same here - this part creates a function (and generates code) on your local machine. This is a separate task. You should remove that

File: automationspn/script.ps1 [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+$resourceGroupName = Read-Host -Prompt "Name your resource group"
+$location = 'West US 2'
+$templateUri = "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/101-function-app-create-dynamic/azuredeploy.json"
+
+#Create new resource group and deploy template
+New-AzResourceGroup -Name $resourceGroupName -Location "$location"
+New-AzResourceGroupDeployment -ResourceGroupName $resourceGroupName -TemplateUri $templateUri
+
+func start
```
Comment (toddysm) [inline automationspn/script.ps1]: This starts the local function (on your machine) and is not needed for the resource automation.

File: automationspn/script.ps1 [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+#create unique name for HTTP Trigger function
+$functionname = Read-Host -Prompt "Name your function"
+#add function to the project
+func new --name $functionname --template "HTTP trigger" --authlevel "anonymous" 
+
+
+$resourceGroupName = Read-Host -Prompt "Name your resource group"
+$location = 'West US 2'
+$templateUri = "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/101-function-app-create-dynamic/azuredeploy.json"
```
Comment (toddysm) [inline automationspn/script.ps1]: I assume this is the URL you mentioned in Slack? This should point to your `azuredeploy.json`. What is the URL you should use?

**Review by toddysm — DISMISSED** [review]

**Review by sunny852206 — APPROVED** [review]
Looks good to me!!

**Review by Pabriones — APPROVED** [review]
saw the changes.. Looks good

**Review by jlsnow301 — APPROVED** [review]
~~Merge conflicts (not your fault in this case) can probably be ignored here~~
merge conflicts are resolved but ds_store files are still there. should be good if you merge upstream/dev

**Pull request merged**